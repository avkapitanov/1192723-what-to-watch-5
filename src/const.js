export const DEFAULT_RATING_FORM_VALUE = 3;
export const MAX_RATING_VALUE_FORM = 5;
export const HOVER_TIME_BEFORE_PLAYING = 2000;
export const FilmTab = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`
};
export const DEFAULT_ACTIVE_TAB = FilmTab.OVERVIEW;
export const DEFAULT_GENRE_FILTER_VALUE = `All genres`;
export const INITIAL_FILMS_COUNT = 8;
export const FILMS_COUNT_PER_STEP = 8;
export const MAX_GENRES_IN_FILTER = 9;
export const API_ENDPOINT = `https://5.react.pages.academy/wtw`;
export const REQUEST_TIMEOUT = 5000;
export const HttpResponseCode = {
  UNAUTHORIZED: 401
};
export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};
export const FilmRating = {
  BAD: {
    VALUE: 3,
    TEXT: `Bad`
  },
  NORMAL: {
    VALUE: 5,
    TEXT: `Normal`
  },
  GOOD: {
    VALUE: 8,
    TEXT: `Good`
  },
  VERY_GOOD: {
    VALUE: 10,
    TEXT: `Very good`
  },
  AWESOME: {
    TEXT: `Awesome`
  }
};

