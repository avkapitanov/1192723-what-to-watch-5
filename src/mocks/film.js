export const promoFilm = {
  "name": `Bronson`,
  "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
  "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
  "background_color": `#73B39A`,
  "description": `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
  "rating": 3.6,
  "scores_count": 109661,
  "director": `Nicolas Winding Refn`,
  "starring": [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
  "run_time": 92,
  "genre": `Action`,
  "released": 2008,
  "id": 1,
  "is_favorite": true,
  "video_link": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
};

export const filmWIthChangedStatus = {
  "name": `Bronson`,
  "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
  "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
  "background_color": `#73B39A`,
  "description": `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
  "rating": 3.6,
  "scores_count": 109661,
  "director": `Nicolas Winding Refn`,
  "starring": [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
  "run_time": 92,
  "genre": `Action`,
  "released": 2008,
  "id": 1,
  "is_favorite": false,
  "video_link": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
};

export const film = {
  id: 1,
  title: `We need to talk about Kevin`,
  genre: [`Drama`],
  year: 2011,
  poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/We_need_to_talk_about_Kevin.jpg`,
  posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/We_need_to_talk_about_Kevin.jpg`,
  background: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/We_need_to_talk_about_Kevin.jpg`,
  video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  backgroundColor: `#E1DFDE`,
  rating: 3.8,
  scoresCount: 123240,
  description: `Kevin's mother struggles to love her strange child, despite the increasingly dangerous things he says and does as he grows up. But Kevin is just getting started, and his final act will be beyond anything anyone imagined.`,
  director: `Lynne Ramsay`,
  starring: [
    `Tilda Swinton`,
    `John C. Reilly`,
    `Ezra Miller`
  ],
  runTime: 112,
  isFavorite: false
};
