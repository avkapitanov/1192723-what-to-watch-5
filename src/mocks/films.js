export const films = [{
  "name": `Bronson`,
  "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
  "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
  "background_color": `#73B39A`,
  "description": `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
  "rating": 3.6,
  "scores_count": 109661,
  "director": `Nicolas Winding Refn`,
  "starring": [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
  "run_time": 92,
  "genre": `Action`,
  "released": 2008,
  "id": 1,
  "is_favorite": true,
  "video_link": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  "name": `Pulp Fiction`,
  "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg`,
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg`,
  "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg`,
  "background_color": `#795433`,
  "description": `The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.`,
  "rating": 1.5,
  "scores_count": 1635992,
  "director": `Quentin Tarantino`,
  "starring": [`John Travolta`, `Uma Thurman`, `Samuel L. Jackson`],
  "run_time": 153,
  "genre": `Crime`,
  "released": 1994,
  "id": 2,
  "is_favorite": true,
  "video_link": `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  "name": `Midnight Special`,
  "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Midnight_Special.jpg`,
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/midnight-special.jpg`,
  "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Midnight_Special.jpg`,
  "background_color": `#828585`,
  "description": `A father and son go on the run, pursued by the government and a cult drawn to the child's special powers.`,
  "rating": 3.3,
  "scores_count": 67815,
  "director": `Jeff Nichols`,
  "starring": [`Michael Shannon`, `Joel Edgerton`, `Kirsten Dunst `],
  "run_time": 112,
  "genre": `Action`,
  "released": 2016,
  "id": 3,
  "is_favorite": true,
  "video_link": `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  "preview_video_link": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
}, {
  "name": `Matrix`,
  "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/matrix.jpg`,
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/matrix.jpg`,
  "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/matrix.jpg`,
  "background_color": `#B9B27E`,
  "description": `A computer hacker learns from mysterious rebels about the true nature of his reality and his role in the war against its controllers.`,
  "rating": 4.4,
  "scores_count": 1503092,
  "director": `Wachowski Brothers`,
  "starring": [`Keanu Reeves`, `Laurence Fishburne`, `Carrie-Anne Moss`],
  "run_time": 136,
  "genre": `Action`,
  "released": 1999,
  "id": 4,
  "is_favorite": true,
  "video_link": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  "name": `Legend`,
  "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Legend.jpg`,
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/legend.jpg`,
  "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/legend.jpg`,
  "background_color": `#E1DAD7`,
  "description": `Identical twin gangsters Ronald and Reginald Kray terrorize London during the 1960s.`,
  "rating": 3.5,
  "scores_count": 138487,
  "director": `Brian Helgeland`,
  "starring": [`Tom Hardy`, `Emily Browning`, `Taron Egerton`],
  "run_time": 132,
  "genre": `Crime`,
  "released": 2015,
  "id": 5,
  "is_favorite": false,
  "video_link": `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  "preview_video_link": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
}];

export const normalizedFilms = {
  "1": {
    id: 1,
    title: `We need to talk about Kevin`,
    genre: [`Drama`],
    year: 2011,
    poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/We_need_to_talk_about_Kevin.jpg`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/We_need_to_talk_about_Kevin.jpg`,
    background: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/We_need_to_talk_about_Kevin.jpg`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    backgroundColor: `#E1DFDE`,
    rating: 3.8,
    scoresCount: 123240,
    description: `Kevin's mother struggles to love her strange child, despite the increasingly dangerous things he says and does as he grows up. But Kevin is just getting started, and his final act will be beyond anything anyone imagined.`,
    director: `Lynne Ramsay`,
    starring: [
      `Tilda Swinton`,
      `John C. Reilly`,
      `Ezra Miller`
    ],
    runTime: 112,
    isFavorite: false
  },
  "2": {
    id: 2,
    title: `What We Do in the Shadows`,
    genre: [`Comedy`, `Drama`],
    year: 2019,
    poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/What-We-Do-in-the-Shadows.jpg`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/What-We-Do-in-the-Shadows.jpg`,
    background: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/What-We-Do-in-the-Shadows.jpg`,
    video: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    backgroundColor: `#A39E81`,
    rating: 7.2,
    scoresCount: 6173,
    description: `A look into the daily (or rather, nightly) lives of three vampires who've lived together for over 100 years, in Staten Island.`,
    director: `Jemaine Clement`,
    starring: [
      `Kayvan Novak`,
      `Matt Berry`,
      `Natasia Demetriou`
    ],
    runTime: 30,
    isFavorite: false
  },
  "3": {
    id: 3,
    title: `Bronson`,
    genre: [`Action`],
    year: 2008,
    poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    background: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    video: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    backgroundColor: `#73B39A`,
    rating: 3.6,
    scoresCount: 109661,
    description: `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
    director: `Nicolas Winding Refn`,
    starring: [
      `Tom Hardy`,
      `Kelly Adams`,
      `Luing Andrews`
    ],
    runTime: 92,
    isFavorite: false
  }
};
