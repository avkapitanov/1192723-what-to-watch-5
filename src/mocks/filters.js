export const filterGenres = [
  `All genres`,
  `Drama`,
  `Comedy`,
  `Horror`,
  `Romance`,
  `Sci-Fi`
];

export const selectedFilterGenre = `Drama`;
