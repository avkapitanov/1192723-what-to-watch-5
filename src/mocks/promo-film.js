const promoFilm = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  year: 2014,
  poster: `./img/the-grand-budapest-hotel-poster.jpg`,
  background: `./img/bg-the-grand-budapest-hotel.jpg`
};

export default promoFilm;
