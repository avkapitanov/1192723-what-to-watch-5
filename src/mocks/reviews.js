import {nanoid} from 'nanoid';
import {getRandomElements, getRandomInteger, generateDate, getRandomFloat} from "../utils";

const REVIEWS_FILM_CARD_COUNT = 9;
const MIN_REVIEW_SENTENCE = 1;
const MAX_REVIEW_SENTENCE = 4;
const MIN_REVIEW_RATING = 6;
const MAX_REVIEW_RATING = 10;

const generateAuthor = () => {
  const authors = [
    `Kate Muir`,
    `Matthew Lickona`,
    `Bill Goodykoontz`,
    `Paula Fleri-Soler`,
    `Amanda Greever`,
    `Cameron Owens`
  ];

  const randomIndex = getRandomInteger(0, authors.length - 1);

  return authors[randomIndex];
};

const generateText = () => {
  const text = [
    `Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight.`,
    `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
    `Fusce tristique felis at fermentum pharetra.`,
    `Aliquam id orci ut lectus varius viverra.`,
    `Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante.`,
    `Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum.`,
    `Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui.`,
    `Sed sed nisi sed augue convallis suscipit in sed felis.`,
    `Aliquam erat volutpat.`,
    `Nunc fermentum tortor ac porta dapibus.`,
    `In rutrum ac purus sit amet tempus.`
  ];

  const sentenceCount = getRandomInteger(MIN_REVIEW_SENTENCE, MAX_REVIEW_SENTENCE);

  return getRandomElements(text, sentenceCount).join(` `);
};

const generateReview = () => {
  const author = generateAuthor();

  return {
    id: nanoid(),
    author,
    rating: getRandomFloat(MIN_REVIEW_RATING, MAX_REVIEW_RATING),
    date: generateDate(),
    text: generateText()
  };
};

const reviews = new Array(REVIEWS_FILM_CARD_COUNT).fill().map(generateReview);

export default reviews;
