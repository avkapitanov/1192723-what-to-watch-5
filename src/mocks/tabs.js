export const tabs = [
  `Overview`,
  `Details`,
  `Reviews`
];

export const activeTab = `Overview`;
