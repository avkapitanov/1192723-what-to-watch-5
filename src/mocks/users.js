export const authInfo = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg`
};
