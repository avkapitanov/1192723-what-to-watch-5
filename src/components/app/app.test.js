import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from 'redux-mock-store';
import {Router as BrowserRouter} from "react-router-dom";
import browserHistory from "../../browser-history";

import {normalizedFilms} from "../../mocks/films";
import {filterGenres} from "../../mocks/filters";
import {authInfo} from "../../mocks/users";

import {AuthorizationStatus, DEFAULT_GENRE_FILTER_VALUE} from "../../const";
import App from "./app";

const mockStore = configureStore([]);
const store = mockStore({
  DATA: {
    films: {
      entities: normalizedFilms
    },
    promoId: 3,
    filterGenres,
  },
  PROCESS: {
    selectedFilterGenre: DEFAULT_GENRE_FILTER_VALUE,
    isLoading: false
  },
  USER: {
    authorizationStatus: AuthorizationStatus.AUTH,
    authInfo
  }
});

it(`Render App`, () => {
  const tree = renderer
      .create(
          <Provider store={store}>
            <BrowserRouter history={browserHistory}>
              <App />
            </BrowserRouter>
          </Provider>
      )
      .toJSON();

  expect(tree).toMatchSnapshot();
});
