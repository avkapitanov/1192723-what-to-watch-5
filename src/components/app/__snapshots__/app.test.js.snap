// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Bronson"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            onClick={[Function]}
            src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg"
            width="63"
          />
        </div>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="Bronson"
            height="327"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Bronson
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Action
            </span>
            <span
              className="movie-card__year"
            >
              2008
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              data-id={3}
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            data-genre="All genres"
            href="#"
            onClick={[Function]}
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            data-genre="Drama"
            href="#"
            onClick={[Function]}
          >
            Drama
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            data-genre="Comedy"
            href="#"
            onClick={[Function]}
          >
            Comedy
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            data-genre="Horror"
            href="#"
            onClick={[Function]}
          >
            Horror
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            data-genre="Romance"
            href="#"
            onClick={[Function]}
          >
            Romance
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            data-genre="Sci-Fi"
            href="#"
            onClick={[Function]}
          >
            Sci-Fi
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <img
              alt="We need to talk about Kevin"
              height="175"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/We_need_to_talk_about_Kevin.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/1"
              onClick={[Function]}
            >
              We need to talk about Kevin
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <img
              alt="What We Do in the Shadows"
              height="175"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/What-We-Do-in-the-Shadows.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/2"
              onClick={[Function]}
            >
              What We Do in the Shadows
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <img
              alt="Bronson"
              height="175"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/3"
              onClick={[Function]}
            >
              Bronson
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
